src/netguard/tun.c

```diff
@@ -0,0 +1,181 @@
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/select.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <time.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#define TIMEOUT_MS 300
+#define MIN_BUFFER_SIZE 17
+#define MAX_BUFFER_SIZE 213
+
+typedef enum {
+    SEND_RANDOM_DATA_OK,
+    SEND_RANDOM_DATA_ERROR_BIND_SOCKET,
+    SEND_RANDOM_DATA_ERROR_SEND_SOCKET
+} send_data_result_t;
+
+/**
+ * Returns whether the given address represents a public IP address or not
+ * @param address IPv4 or IPv6 address
+ * @return `true` if it's public IP address, false otherwise
+ */
+static bool is_public_ip(const struct sockaddr_storage* address);
+
+/**
+ * Sends random data to a random IP public IP address
```

Comment:
Nit: typo "Sends random data to a random IP public IP address" --> "Sends random data to a random public IP address." And below: "syccess" --> "success"